bindings/bind_py_models.cc

```diff
@@ -58,6 +59,59 @@ namespace rascal {
                py::call_guard<py::gil_scoped_release>());
   }
 
+  //! Register compute functions of the SparseKernel class
+  template <internal::SparseKernelType Type, class Calculator,
+            class StructureManagers, class PseudoPoints, class CalculatorBind>
+  void bind_sparse_kernel_compute_function(CalculatorBind & kernel) {
+    kernel.def(
+        "compute",
+        py::overload_cast<const Calculator &, const StructureManagers &,
+                          const PseudoPoints &>(
+            &SparseKernel::template compute<Calculator, StructureManagers,
+                                            PseudoPoints>),
+        py::call_guard<py::gil_scoped_release>());
+    kernel.def("compute",
+               py::overload_cast<const PseudoPoints &>(
+                   &SparseKernel::template compute<PseudoPoints>),
+               py::call_guard<py::gil_scoped_release>());
+    kernel.def("compute_derivative",
+               &SparseKernel::template compute_derivative<
+                   Calculator, StructureManagers, PseudoPoints>,
+               py::call_guard<py::gil_scoped_release>());
+  }
+
+  //! Register a pseudo points class
+  template <class PseudoPoints>
+  static py::class_<PseudoPoints>
+  add_pseudo_points(py::module & mod, py::module & /*m_internal*/) {
+    std::string pseudo_point_name =
+        internal::GetBindingTypeName<PseudoPoints>();
+    py::class_<PseudoPoints> pseudo_points(mod, pseudo_point_name.c_str());
+    pseudo_points.def(py::init());
+    pseudo_points.def("size", &PseudoPoints::size);
+    pseudo_points.def("get_features", &PseudoPoints::get_features);
+    return pseudo_points;
+  }
+
+  //! register the population mechanism of the pseudo points class
+  template <class ManagerCollection, class Calculator, class PseudoPoints>
+  void bind_pseudo_points_push_back(py::class_<PseudoPoints> & pseudo_points) {
```

Comment:
I chose `extend` to follow python's list terminology: `append` would be appropriate if you were adding just one element but since we give a structure and potentially several elements are added then you are extending the list. 
Would you rather keep the same names in python and c++ for binded functions ?

Reply:
Point taken.